test/delegations.spec.ts

```diff
@@ -301,123 +301,129 @@ describe('delegations-contract', async () => {
-       await d2.stake(200);
-
-       await d.contractRegistry.setContract("delegations", d.delegations.address, true, {from: d.registryAdmin.address});
-
-        const {v: v1} = await d.newGuardian(100, false, false, true);
-        const {v: v2} = await d.newGuardian(100, false, false, true);
-
-       let r = await d.delegations.importDelegations([d1.address, d2.address], [v1.address, v2.address], false, {from: d.migrationManager.address});
-       expect(r).to.have.a.delegationsImportedEvent({
-           from: [d1.address, d2.address],
-           to: [v1.address, v2.address]
-       });
-       expect(r).to.have.a.delegatedEvent({from: d1.address, to: v1.address});
-       expect(r).to.have.a.delegatedStakeChangedEvent({
-           addr: v1.address,
-           delegatedStake: bn(200),
-           delegators: [d1.address],
-           delegatorTotalStakes: [bn(100)]
-       });
-       expect(r).to.have.a.delegatedEvent({from: d2.address, to: v2.address});
-       expect(r).to.have.a.delegatedStakeChangedEvent({
-           addr: v2.address,
-           delegatedStake: bn(300),
-           delegators: [d2.address],
-           delegatorTotalStakes: [bn(200)]
-       });
-       expect(r).to.not.have.a.committeeSnapshotEvent();
-
-       // import a delegation when already delegated
-
-       r = await d.delegations.importDelegations([d1.address, d2.address], [v2.address, v1.address], false, {from: d.migrationManager.address});
-       expect(r).to.have.a.delegationsImportedEvent({
-            from: [d1.address, d2.address],
-            to: [v2.address, v1.address]
-        });
-       expect(r).to.have.a.delegatedEvent({from: d2.address, to: v1.address});
-       expect(r).to.have.a.delegatedStakeChangedEvent({
-           addr: v1.address,
-           delegatedStake: bn(300),
-       });
-       expect(r).to.have.a.delegatedEvent({from: d1.address, to: v2.address});
-       expect(r).to.have.a.delegatedStakeChangedEvent({
-           addr: v2.address,
-           delegatedStake: bn(200),
-       });
-       expect(r).to.not.have.a.committeeSnapshotEvent();
-
-    });
-
-    it('imports a delegation for a delegator with an existing stake (with election notification)', async () => {
-       const d = await Driver.new();
-
-       const otherDelegationContract = await d.web3.deploy("Delegations", [d.contractRegistry.address, d.registryAdmin.address], null, d.session);
-
-       await d.contractRegistry.setContract("delegations", otherDelegationContract.address, true, {from: d.registryAdmin.address});
+    const importDelegationsTestGenerator = (notifyElections: boolean) => {
+        return async () => {
+            const d = await Driver.new();
+
+            // setup a throwaway contract to prevent staking notifications from reaching the real Delegations contract
+            const notificationsSwallower = await d.web3.deploy("Delegations", [d.contractRegistry.address, d.registryAdmin.address], null, d.session);
```

Comment:
`StakingContractHandler` now has a method to enable/disable notifications, this can be used instead